// huawei-lte-api/dist/enums/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseCodeEnum = exports.ResponseEnum = void 0;
var ResponseEnum;
(function (ResponseEnum) {
    ResponseEnum["OK"] = "OK";
})(ResponseEnum = exports.ResponseEnum || (exports.ResponseEnum = {}));
var ResponseCodeEnum;
(function (ResponseCodeEnum) {
    ResponseCodeEnum[ResponseCodeEnum["ERROR_SYSTEM_UNKNOWN"] = 100001] = "ERROR_SYSTEM_UNKNOWN";
    ResponseCodeEnum[ResponseCodeEnum["ERROR_SYSTEM_NO_SUPPORT"] = 100002] = "ERROR_SYSTEM_NO_SUPPORT";
    ResponseCodeEnum[ResponseCodeEnum["ERROR_SYSTEM_NO_RIGHTS"] = 100003] = "ERROR_SYSTEM_NO_RIGHTS";
    ResponseCodeEnum[ResponseCodeEnum["ERROR_SYSTEM_BUSY"] = 100004] = "ERROR_SYSTEM_BUSY";
    ResponseCodeEnum[ResponseCodeEnum["ERROR_FORMAT_ERROR"] = 100005] = "ERROR_FORMAT_ERROR";
    ResponseCodeEnum[ResponseCodeEnum["ERROR_SYSTEM_CSRF"] = 125002] = "ERROR_SYSTEM_CSRF";
})(ResponseCodeEnum = exports.ResponseCodeEnum || (exports.ResponseCodeEnum = {}));
