// huawei-lte-api/dist/enums/sms.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriorityEnum = exports.SendTypeEnum = exports.SaveModeEnum = exports.TextModeEnum = exports.BoxTypeEnum = void 0;
var BoxTypeEnum;
(function (BoxTypeEnum) {
    BoxTypeEnum[BoxTypeEnum["LOCAL_INBOX"] = 1] = "LOCAL_INBOX";
    BoxTypeEnum[BoxTypeEnum["LOCAL_SENT"] = 2] = "LOCAL_SENT";
    BoxTypeEnum[BoxTypeEnum["LOCAL_DRAFT"] = 3] = "LOCAL_DRAFT";
    BoxTypeEnum[BoxTypeEnum["LOCAL_TRASH"] = 4] = "LOCAL_TRASH";
    BoxTypeEnum[BoxTypeEnum["SIM_INBOX"] = 5] = "SIM_INBOX";
    BoxTypeEnum[BoxTypeEnum["SIM_SENT"] = 6] = "SIM_SENT";
    BoxTypeEnum[BoxTypeEnum["SIM_DRAFT"] = 7] = "SIM_DRAFT";
    BoxTypeEnum[BoxTypeEnum["MIX_INBOX"] = 8] = "MIX_INBOX";
    BoxTypeEnum[BoxTypeEnum["MIX_SENT"] = 9] = "MIX_SENT";
    BoxTypeEnum[BoxTypeEnum["MIX_DRAFT"] = 10] = "MIX_DRAFT";
})(BoxTypeEnum = exports.BoxTypeEnum || (exports.BoxTypeEnum = {}));
var TextModeEnum;
(function (TextModeEnum) {
    TextModeEnum[TextModeEnum["UCS2"] = 0] = "UCS2";
    TextModeEnum[TextModeEnum["SEVEN_BIT"] = 1] = "SEVEN_BIT";
    TextModeEnum[TextModeEnum["EIGHT_BIT"] = 2] = "EIGHT_BIT";
})(TextModeEnum = exports.TextModeEnum || (exports.TextModeEnum = {}));
var SaveModeEnum;
(function (SaveModeEnum) {
    SaveModeEnum[SaveModeEnum["LOCAL"] = 0] = "LOCAL";
    SaveModeEnum[SaveModeEnum["SIM_CARD"] = 1] = "SIM_CARD";
    SaveModeEnum[SaveModeEnum["SIM_CARD_FIRST"] = 2] = "SIM_CARD_FIRST";
    SaveModeEnum[SaveModeEnum["LOCAL_FIRST"] = 3] = "LOCAL_FIRST";
})(SaveModeEnum = exports.SaveModeEnum || (exports.SaveModeEnum = {}));
var SendTypeEnum;
(function (SendTypeEnum) {
    SendTypeEnum[SendTypeEnum["SEND"] = 0] = "SEND";
    SendTypeEnum[SendTypeEnum["SEND_AND_SAVE"] = 1] = "SEND_AND_SAVE";
})(SendTypeEnum = exports.SendTypeEnum || (exports.SendTypeEnum = {}));
var PriorityEnum;
(function (PriorityEnum) {
    PriorityEnum[PriorityEnum["NORMAL"] = 0] = "NORMAL";
    PriorityEnum[PriorityEnum["INTERACTIVE"] = 1] = "INTERACTIVE";
    PriorityEnum[PriorityEnum["URGENT"] = 2] = "URGENT";
    PriorityEnum[PriorityEnum["EMERGENCY"] = 3] = "EMERGENCY";
})(PriorityEnum = exports.PriorityEnum || (exports.PriorityEnum = {}));
