// huawei-lte-api/dist/enums/user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionErrorEnum = exports.LoginStateEnum = exports.LoginErrorEnum = exports.PasswordTypeEnum = void 0;
var PasswordTypeEnum;
(function (PasswordTypeEnum) {
    PasswordTypeEnum[PasswordTypeEnum["BASE_64"] = 0] = "BASE_64";
    PasswordTypeEnum[PasswordTypeEnum["BASE_64_AFTER_PASSWORD_CHANGE"] = 3] = "BASE_64_AFTER_PASSWORD_CHANGE";
    PasswordTypeEnum[PasswordTypeEnum["SHA256"] = 4] = "SHA256";
})(PasswordTypeEnum = exports.PasswordTypeEnum || (exports.PasswordTypeEnum = {}));
var LoginErrorEnum;
(function (LoginErrorEnum) {
    LoginErrorEnum[LoginErrorEnum["USERNAME_WRONG"] = 108001] = "USERNAME_WRONG";
    LoginErrorEnum[LoginErrorEnum["PASSWORD_WRONG"] = 108002] = "PASSWORD_WRONG";
    LoginErrorEnum[LoginErrorEnum["ALREADY_LOGIN"] = 108003] = "ALREADY_LOGIN";
    LoginErrorEnum[LoginErrorEnum["USERNAME_PWD_WRONG"] = 108006] = "USERNAME_PWD_WRONG";
    LoginErrorEnum[LoginErrorEnum["USERNAME_PWD_ORERRUN"] = 108007] = "USERNAME_PWD_ORERRUN";
    LoginErrorEnum[LoginErrorEnum["USERNAME_PWD_MODIFY"] = 115002] = "USERNAME_PWD_MODIFY";
})(LoginErrorEnum = exports.LoginErrorEnum || (exports.LoginErrorEnum = {}));
var LoginStateEnum;
(function (LoginStateEnum) {
    LoginStateEnum[LoginStateEnum["LOGGED_IN"] = 0] = "LOGGED_IN";
    LoginStateEnum[LoginStateEnum["LOGGED_OUT"] = -1] = "LOGGED_OUT";
    LoginStateEnum[LoginStateEnum["REPEAT"] = -2] = "REPEAT";
})(LoginStateEnum = exports.LoginStateEnum || (exports.LoginStateEnum = {}));
var SessionErrorEnum;
(function (SessionErrorEnum) {
    SessionErrorEnum[SessionErrorEnum["VOICE_BUSY"] = 120001] = "VOICE_BUSY";
    SessionErrorEnum[SessionErrorEnum["WRONG_TOKEN"] = 125001] = "WRONG_TOKEN";
    SessionErrorEnum[SessionErrorEnum["WRONG_SESSION"] = 125002] = "WRONG_SESSION";
    SessionErrorEnum[SessionErrorEnum["WRONG_SESSION_TOKEN"] = 125003] = "WRONG_SESSION_TOKEN";
})(SessionErrorEnum = exports.SessionErrorEnum || (exports.SessionErrorEnum = {}));
