// huawei-lte-api/dist/enums/wlan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WepEncryptModeEnum = exports.WpaEncryptModeEnum = exports.AuthModeEnum = void 0;
var AuthModeEnum;
(function (AuthModeEnum) {
    AuthModeEnum["AUTO"] = "AUTO";
    AuthModeEnum["OPEN"] = "OPEN";
    AuthModeEnum["SHARE"] = "SHARE";
    AuthModeEnum["WPA_PSK"] = "WPA-PSK";
    AuthModeEnum["WPA2_PSK"] = "WPA2-PSK";
    AuthModeEnum["WPA_WPA2_PSK"] = "WPA/WPA2-PSK";
})(AuthModeEnum = exports.AuthModeEnum || (exports.AuthModeEnum = {}));
var WpaEncryptModeEnum;
(function (WpaEncryptModeEnum) {
    WpaEncryptModeEnum["AES"] = "AES";
    WpaEncryptModeEnum["TKIP"] = "TKIP";
    WpaEncryptModeEnum["MIX"] = "MIX";
})(WpaEncryptModeEnum = exports.WpaEncryptModeEnum || (exports.WpaEncryptModeEnum = {}));
var WepEncryptModeEnum;
(function (WepEncryptModeEnum) {
    WepEncryptModeEnum["NONE"] = "NONE";
    WepEncryptModeEnum["WEP"] = "WEP";
    WepEncryptModeEnum["WEP64"] = "WEP64";
    WepEncryptModeEnum["WEP128"] = "WEP128";
})(WepEncryptModeEnum = exports.WepEncryptModeEnum || (exports.WepEncryptModeEnum = {}));
