// huawei-lte-api/dist/exceptions.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestFormatException = exports.LoginErrorUsernamePasswordModifyException = exports.LoginErrorUsernamePasswordOverrunException = exports.LoginErrorUsernamePasswordWrongException = exports.LoginErrorAlreadyLoginException = exports.LoginErrorPasswordWrongException = exports.LoginErrorUsernameWrongException = exports.ResponseErrorLoginCsrfException = exports.ResponseErrorSystemBusyException = exports.ResponseErrorLoginRequiredException = exports.ResponseErrorNotSupportedException = exports.ResponseErrorException = void 0;
var ResponseErrorException = /** @class */ (function (_super) {
    __extends(ResponseErrorException, _super);
    function ResponseErrorException(message, code) {
        var _this = _super.call(this, message) || this;
        _this.code = code;
        Object.setPrototypeOf(_this, ResponseErrorException.prototype);
        return _this;
    }
    return ResponseErrorException;
}(Error));
exports.ResponseErrorException = ResponseErrorException;
var ResponseErrorNotSupportedException = /** @class */ (function (_super) {
    __extends(ResponseErrorNotSupportedException, _super);
    function ResponseErrorNotSupportedException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ResponseErrorNotSupportedException;
}(ResponseErrorException));
exports.ResponseErrorNotSupportedException = ResponseErrorNotSupportedException;
var ResponseErrorLoginRequiredException = /** @class */ (function (_super) {
    __extends(ResponseErrorLoginRequiredException, _super);
    function ResponseErrorLoginRequiredException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ResponseErrorLoginRequiredException;
}(ResponseErrorException));
exports.ResponseErrorLoginRequiredException = ResponseErrorLoginRequiredException;
var ResponseErrorSystemBusyException = /** @class */ (function (_super) {
    __extends(ResponseErrorSystemBusyException, _super);
    function ResponseErrorSystemBusyException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ResponseErrorSystemBusyException;
}(ResponseErrorException));
exports.ResponseErrorSystemBusyException = ResponseErrorSystemBusyException;
var ResponseErrorLoginCsrfException = /** @class */ (function (_super) {
    __extends(ResponseErrorLoginCsrfException, _super);
    function ResponseErrorLoginCsrfException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ResponseErrorLoginCsrfException;
}(ResponseErrorException));
exports.ResponseErrorLoginCsrfException = ResponseErrorLoginCsrfException;
var LoginErrorUsernameWrongException = /** @class */ (function (_super) {
    __extends(LoginErrorUsernameWrongException, _super);
    function LoginErrorUsernameWrongException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorUsernameWrongException;
}(ResponseErrorException));
exports.LoginErrorUsernameWrongException = LoginErrorUsernameWrongException;
var LoginErrorPasswordWrongException = /** @class */ (function (_super) {
    __extends(LoginErrorPasswordWrongException, _super);
    function LoginErrorPasswordWrongException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorPasswordWrongException;
}(ResponseErrorException));
exports.LoginErrorPasswordWrongException = LoginErrorPasswordWrongException;
var LoginErrorAlreadyLoginException = /** @class */ (function (_super) {
    __extends(LoginErrorAlreadyLoginException, _super);
    function LoginErrorAlreadyLoginException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorAlreadyLoginException;
}(ResponseErrorException));
exports.LoginErrorAlreadyLoginException = LoginErrorAlreadyLoginException;
var LoginErrorUsernamePasswordWrongException = /** @class */ (function (_super) {
    __extends(LoginErrorUsernamePasswordWrongException, _super);
    function LoginErrorUsernamePasswordWrongException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorUsernamePasswordWrongException;
}(ResponseErrorException));
exports.LoginErrorUsernamePasswordWrongException = LoginErrorUsernamePasswordWrongException;
var LoginErrorUsernamePasswordOverrunException = /** @class */ (function (_super) {
    __extends(LoginErrorUsernamePasswordOverrunException, _super);
    function LoginErrorUsernamePasswordOverrunException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorUsernamePasswordOverrunException;
}(ResponseErrorException));
exports.LoginErrorUsernamePasswordOverrunException = LoginErrorUsernamePasswordOverrunException;
var LoginErrorUsernamePasswordModifyException = /** @class */ (function (_super) {
    __extends(LoginErrorUsernamePasswordModifyException, _super);
    function LoginErrorUsernamePasswordModifyException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LoginErrorUsernamePasswordModifyException;
}(ResponseErrorException));
exports.LoginErrorUsernamePasswordModifyException = LoginErrorUsernamePasswordModifyException;
var RequestFormatException = /** @class */ (function (_super) {
    __extends(RequestFormatException, _super);
    function RequestFormatException() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return RequestFormatException;
}(ResponseErrorException));
exports.RequestFormatException = RequestFormatException;
