// huawei-lte-api/dist/api/Net.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Net = void 0;
var ApiGroup_1 = require("../ApiGroup");
var Net = /** @class */ (function (_super) {
    __extends(Net, _super);
    function Net() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Net.prototype.currentPlmn = function () {
        return this._connection.get('net/current-plmn');
    };
    Net.prototype.netMode = function () {
        return this._connection.get('net/net-mode');
    };
    Net.prototype.setNetMode = function (lteband, networkband, networkmode) {
        return this._connection.postSet('net/net-mode', {
            'NetworkMode': networkmode,
            'NetworkBand': networkband,
            'LTEBand': lteband
        });
    };
    Net.prototype.network = function () {
        return this._connection.get('net/network');
    };
    Net.prototype.register = function () {
        return this._connection.get('net/register');
    };
    /**
     * Sets network selection
     * @param mode "1": manual network selection, "0": auto
     * @param plmn Plmn code ("Numeric" value returned by net_mode_list()), "" for auto
     * @param rat "0": "2G", "2": "3G", "7": "4G" ("Rat" value returned by net_mode_list()), "" for auto
     */
    Net.prototype.setRegister = function (mode, plmn, rat) {
        return this._connection.postSet('net/register', {
            'Mode': mode,
            'Plmn': plmn,
            'Rat': rat
        });
    };
    Net.prototype.netModeList = function () {
        return this._connection.get('net/net-mode-list');
    };
    /**
     * DoS
     */
    Net.prototype.plmnList = function () {
        return this._connection.get('net/plmn-list');
    };
    Net.prototype.netFeatureSwitch = function () {
        return this._connection.get('net/net-feature-switch');
    };
    Net.prototype.cellInfo = function () {
        return this._connection.get('net/cell-info');
    };
    Net.prototype.cspsState = function () {
        return this._connection.get('net/csps_state');
    };
    return Net;
}(ApiGroup_1.ApiGroup));
exports.Net = Net;
