// huawei-lte-api/dist/api/OnlineUpdate.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnlineUpdate = void 0;
var ApiGroup_1 = require("../ApiGroup");
var OnlineUpdate = /** @class */ (function (_super) {
    __extends(OnlineUpdate, _super);
    function OnlineUpdate() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    OnlineUpdate.prototype.checkNewVersion = function () {
        return this._connection.get('online-update/check-new-version');
    };
    OnlineUpdate.prototype.setCheckNewVersion = function () {
        return this._connection.postSet('online-update/check-new-version', {});
    };
    OnlineUpdate.prototype.status = function () {
        return this._connection.get('online-update/status');
    };
    OnlineUpdate.prototype.urlList = function () {
        return this._connection.get('online-update/url-list');
    };
    OnlineUpdate.prototype.ackNewversion = function () {
        return this._connection.get('online-update/ack-newversion');
    };
    OnlineUpdate.prototype.setAckNewversion = function () {
        return this._connection.postSet('online-update/ack-newversion', {
            'userAckNewVersion': 0
        });
    };
    /**
     * Invoking this method is known to cause some devices to reboot.
     */
    OnlineUpdate.prototype.cancelDownloading = function () {
        return this._connection.get('online-update/cancel-downloading');
    };
    OnlineUpdate.prototype.setCancelDownloading = function () {
        return this._connection.postSet('online-update/cancel-downloading', {});
    };
    OnlineUpdate.prototype.upgradeMessagebox = function () {
        return this._connection.get('online-update/upgrade-messagebox');
    };
    OnlineUpdate.prototype.setUpgradeMessagebox = function (messagebox) {
        return this._connection.postSet('online-update/upgrade-messagebox', {
            'messagebox': messagebox
        });
    };
    OnlineUpdate.prototype.configuration = function () {
        return this._connection.get('online-update/configuration');
    };
    OnlineUpdate.prototype.autoupdateConfig = function () {
        return this._connection.get('online-update/autoupdate-config');
    };
    OnlineUpdate.prototype.redirectCancel = function () {
        return this._connection.get('online-update/redirect_cancel');
    };
    return OnlineUpdate;
}(ApiGroup_1.ApiGroup));
exports.OnlineUpdate = OnlineUpdate;
