// huawei-lte-api/dist/api/Vpn.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vpn = void 0;
var ApiGroup_1 = require("../ApiGroup");
var Vpn = /** @class */ (function (_super) {
    __extends(Vpn, _super);
    function Vpn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Vpn.prototype.featureSwitch = function () {
        return this._connection.get('vpn/feature-switch');
    };
    Vpn.prototype.brList = function () {
        return this._connection.get('vpn/br_list');
    };
    Vpn.prototype.ipsecSettings = function () {
        return this._connection.get('vpn/ipsec_settings');
    };
    Vpn.prototype.l2tpSettings = function () {
        return this._connection.get('vpn/l2tp_settings');
    };
    Vpn.prototype.pptpSettings = function () {
        return this._connection.get('vpn/pptp_settings');
    };
    /**
     * Endpoint found by reverse engineering B310s-22 firmware, unknown usage
     */
    Vpn.prototype.status = function () {
        return this._connection.get('vpn/status');
    };
    return Vpn;
}(ApiGroup_1.ApiGroup));
exports.Vpn = Vpn;
