// huawei-lte-api/dist/config/Network.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Network = void 0;
var ApiGroup_1 = require("../ApiGroup");
var Network = /** @class */ (function (_super) {
    __extends(Network, _super);
    function Network() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Network.prototype.netMode = function () {
        return this._connection.get('network/net-mode.xml', {}, 'config');
    };
    Network.prototype.networkmode = function () {
        return this._connection.get('network/networkmode.xml', {}, 'config');
    };
    Network.prototype.config = function () {
        return this._connection.get('network/config.xml', {}, 'config');
    };
    Network.prototype.networkbandNull = function () {
        return this._connection.get('network/networkband_null.xml', {}, 'config');
    };
    Network.prototype.setOnly4g = function () {
        return this._connection.get('network/setOnly4g.xml', {}, 'config');
    };
    return Network;
}(ApiGroup_1.ApiGroup));
exports.Network = Network;
